### PR TITLE
Upgrade to JBX 2.0.0 and extend lock changes

### PR DESCRIPTION
**Pull request by xBA5ED**
- Upgraded npm package to newest version of JBX contracts
- Did some refactoring to get the tests working with the new contracts
- Burn old and mint new tokenId on extendLock
- Added some sections such as 'private helper functions' etc.
- Moved the specifications packing into a method to simplify the code and prevent accidental differences between them
- Changed `_holder` in `redeemTokensOf ` (see comments)
- Added modifier to `redeem(...)` (see comments)

### CODE REVIEW

File: contracts/JBveBanny.sol [hunk 1]
```diff
@@ -355,9 +370,10 @@ contract JBveBanny is ERC721Votes, ERC721Enumerable, Ownable, ReentrancyGuard, J
 
     // Redeem the locked tokens to reclaim treasury funds.
     uint256 _reclaimedAmount = _terminal.redeemTokensOf(
-      _owner,
+      address(this),
```
Comment (xBA5ED) [inline contracts/JBveBanny.sol:373]: This contract holds the user's tokens right?

File: contracts/JBveBanny.sol [hunk 2]
```diff
@@ -344,6 +354,11 @@ contract JBveBanny is ERC721Votes, ERC721Enumerable, Ownable, ReentrancyGuard, J
     bytes memory _metadata,
     IJBPayoutRedemptionPaymentTerminal _terminal
   ) external nonReentrant {
+    {
+      // Check the permissions scoped to prevent stack too deep
+      _requirePermission(ownerOf(_tokenId), projectId, JBStakingOperations.REDEEM);
+    }
```
Comment (xBA5ED) [inline contracts/JBveBanny.sol:360]: I believe any user could redeem any other user's tokens since the caller wasn't getting checked

Comment (mejango) [reply contracts/JBveBanny.sol:360]: the permission check happens in Terminal. The token holder has to give the veBanny operator permission to redeem its tokens on its behalf.

Comment (mejango) [reply contracts/JBveBanny.sol:360]: worth a comment explaining this inline

File: contracts/JBveBanny.sol (hunk 1 above)
Comment (mejango) [reply contracts/JBveBanny.sol:373]: oh, nice. ok then disregard the previous comment. We need a permissions check, but maybe use a modifier?

**Review by mejango — COMMENTED** [review]
This is great.

There are a handful of open PR's right now that need consolidation to prevent a mess. I'll try to get this done so we can make progress with this PR.

**Comment by mejango** [comment]
Ok the `master` branch is most up-to-date. Can you handle conflicts then push up an updated version of this PR? We'll merge it in right away and work from there.

**Pull request merged**